packages/client/auth-core/src/services/CrossmintService.ts

```diff
@@ -0,0 +1,59 @@
+import { CROSSMINT_PROD_URL, CROSSMINT_STG_URL, CrossmintEnvironment } from "../utils";
+
+export type FetchCrossmintParams = {
+    endpoint: string;
+    options: {
+        method: "GET" | "POST" | "PUT" | "DELETE";
+        body?: any;
+    };
+    onServerErrorMessage?: string;
+};
+
+export class CrossmintService {
+    protected crossmintAPIHeaders: Record<string, string>;
+    public readonly crossmintBaseUrl: string;
+    private static urlMap: Record<CrossmintEnvironment, string> = {
+        staging: CROSSMINT_STG_URL,
+        production: CROSSMINT_PROD_URL,
+    };
+
+    constructor(apiKey: string, jwtToken: string | null, environment?: CrossmintEnvironment) {
```

Comment:
you're not validating the api key, which can be done with `validateAPIKey(apiKey)`, the return of this tells you what environment the apikey is for - so you should use that and drop the environment param completely

Reply:
actually, its a big smell that this code is nearly exactly the same as the CrossmintService in the aa sdk

which is why i made the `ApiClient` and `CrossmintApiClient` classes in the common sdk, pls use those classes! rather than having 3 different services that do ~the same thing, and all of us working separately on them, we should be centralizing it to these new classes

see this pr https://github.com/Crossmint/crossmint-sdk/pull/609/files

Reply:
this is a great point, will address in a separate PR though to unblock
[WAL-2620 : Update `CrossmintService` to use `CrossmintApiClient`](https://linear.app/crossmint/issue/WAL-2620/update-crossmintservice-to-use-crossmintapiclient)